src/UI/admin/components/hall-reservation/AdminHallReservation.js

```diff
@@ -0,0 +1,272 @@
+          </TableWrapper>
+
+          <Info>
+            <InfoTitle>
+              {
+                //this.state.dummyDataArray[0].event
+                //this.state.dummyDataArray[0].hallReservaltions[0]
+                //.reservationTitle
+                //this.state.currentReservationActive.reservationTitle === undefined ? "WWWWWWWW" : this.state.currentReservationActive.reservationTitle
+                this.state.currentReservationActive.reservationTitle
+                //this.state.currentReservationActive
+              }
+            </InfoTitle>
+            <InfoTime>
+              {//this.state.dummyDataArray[0].startTime.toString().substring(4, 25)
+              this.state.currentReservationActive.reservationDate +
+                ' ' +
+                this.state.currentReservationActive.reservationStartTime.slice(
+                  0,
+                  -3
+                )}{' '}
+              -
+              {//this.state.dummyDataArray[0].endTime.toString().substring(4, 25)
+              this.state.currentReservationActive.reservationEndTime.slice(
+                0,
+                -3
+              )}
+            </InfoTime>
+            <Details
+              img={DetailsIcon}
+              onClick={() => {
+                //console.log(this.state.currentReservationActive.reservationDescription);
+                //this.state.currentReservationActive.infoVisibility = 'visible';
+                this.setState({
+                  currentReservationActive: {
+                    reservationTitle: this.state.currentReservationActive
+                      .reservationTitle,
+                    reservationDate: this.state.currentReservationActive
+                      .reservationDate,
+                    reservationStartTime: this.state.currentReservationActive
+                      .reservationStartTime,
+                    reservationEndTime: this.state.currentReservationActive
+                      .reservationEndTime,
+                    reservationDescription: this.state.currentReservationActive
+                      .reservationDescription,
+                    infoVisibility: 'visible'
+                  }
+                });
+
+                console.log(
+                  'currentReservationActive' +
+                    this.state.currentReservationActive
+                );
+              }}
+            >
+              <PopUpInfo
+                style={{
+                  visibility: this.state.currentReservationActive.infoVisibility
+                }}
+              >
+                <PopUpInfoText>
+                  {this.state.currentReservationActive.reservationDescription}
+                </PopUpInfoText>
+              </PopUpInfo>
+            </Details>
+
+            <InfoEraseButton>
+              <InfoEraseButtonText>IZBRIŠI</InfoEraseButtonText>
+            </InfoEraseButton>
+          </Info>
+        </TableAndInfoWrapper>
+      </MainWrapper>
+    );
+  }
+}
```

Comment:
uff... skoro sam umro dešifrirajući ovaj špageti kod :D molim te razbij to u manje cjeline prema uputama... inače, što se tiče funkcionalnosti i logike, nice job